models/articles.models.js

```diff
@@ -1,19 +1,71 @@
 const db = require("../db/connection");
 const format = require("pg-format");
 
-exports.selectArticles = () => {
-  return db
-    .query(
-      `SELECT articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, article_img_url, COUNT(comment_id) AS comment_count
-    FROM articles 
-    LEFT JOIN comments ON comments.article_id = articles.article_id 
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC
-    `
-    )
-    .then(({ rows }) => {
-      return rows;
-    });
+exports.selectArticles = (
+  topic = null,
+  sort_by = "articles.created_at",
+  order = "DESC"
+) => {
+  let query = `SELECT articles.author, title, articles.article_id, topic, articles.created_at, articles.votes, article_img_url, COUNT(comment_id) AS comment_count
+  FROM articles 
+  LEFT JOIN comments ON comments.article_id = articles.article_id 
+  `;
+
+  const validQueries = [
+    "articles.author",
+    "title",
+    "articles.article_id",
+    "topic",
+    "articles.created_at",
+    "articles.votes",
+    "comment_count",
+    "DESC",
+    "ASC",
+  ];
+
+  if (sort_by === "author") {
+    sort_by = validQueries[0];
+  } else if (sort_by === "id") {
+    sort_by = validQueries[2];
+  } else if (sort_by === "date") {
+    sort_by = validQueries[4];
+  } else if (sort_by === "votes") {
+    sort_by = validQueries[5];
+  }
+
+  const validTopics = ["mitch", "cats", "coding", "football", "cooking", null];
```

Comment:
This does work in this case, It might be a nice refactor once youre finished the tickets to come back and make this dynamic (getting the topics from the DB) just incase a topic ever got added ect.